### ISSUE TITLE
Last requested info not reset?

### ISSUE THREAD

**Issue by angerman** [slact/nchan#114]
Maybe I'm misunderstanding what `last requested` is supposed to mean. But I'm seeing:
```
queued messages: 0
last requested: 318 sec. ago (-1=never)
active subscribers: 8
```
despite me sending data continuously and adding new clients.

I did set:
```
nchan_use_redis off;
nchan_storage_engine memory;
nchan_max_reserved_memory 64M;
nchan_store_messages off;
nchan_subscriber_timeout 300;
```
and
```
   location ~ /sub/(\w+)$ {
      nchan_subscriber;
      nchan_channel_id $1;
   }

   location ~ /pub/(\w+)$ {
      nchan_publisher;
      nchan_channel_id $1;
   }
```

at some point all subscriber just disappear and the last request is reset to `0`.

It this supposed to happen? I expected each subscriber to listen up to 300
seconds until a reconnect was required.

**Comment by slact** [comment]
To answer the first part, 'last requested' should be reset to 0 after each new subscriber request for a message. So if you have a longpoll subscriber, it'll reset whenever you send a request, and for a websocket subscriber, it will reset every time the sub received a message (because the next message is automatically requested internally). Is this not the behaviour you're seeing?

The second part: 
>  at some point all subscriber just disappear and the last request is reset to 0

Does this happen after the configured 300 seconds? That's how it should work. Once all subscribers disconnect, and all messages have expires, a channel is deleted within 5-10 seconds.

Let me know and I'll fix it if it's broken.

**Comment by slact** [comment]
I checked it anyway and you're right, last requested is not being updated for eventsource and websocket automatic requests. This will be fixed tomorrow... maybe today.

But I'm more concerned about the second thing -- were clients themselves disappearing (disconnecting) before the configured 300sec time?

**Comment by angerman** [comment]
Thanks. I'll monitor it and report back if I have something more substantial.

Cheers,
 Moritz

**Comment by slact** [comment]
I've fixed this in 643607e4a8c44e53bd90651d3ce5063bdb51aab1 . Feel free to try it out, and thanks for your bug report.
